README.md

```diff
@@ -94,9 +94,8 @@ let router = Router::builder(endpoint)
 struct Echo;
 
 impl ProtocolHandler for Echo {
-    fn accept(self: Arc<Self>, connecting: Connecting) -> BoxedFuture<Result<()>> {
+    fn accept(&self, connection: Connection) -> BoxedFuture<Result<()>> {
```

Comment:
Why was this Arc<Self> before? Was the readme not up to date?

Reply:
> Why was this Arc before? Was the readme not up to date?

indeed

Reply:
Hm. How do we make sure the readme's code is doc-tested?

Reply:
`include_str!` it into a doc comment on a private dummy item and then `cargo doc` might pick it up?  (not tested if a private item is enough for this).

Reply:
Can confirm that flub's suggestion works. I had a few of these trigger errors in our protocol repos that contain the readme as part of the documentation, when APIs changed and no longer compiled.